lib/Mojo/Server/Daemon.pm

```diff
@@ -35,6 +35,7 @@ sub run {
   my $loop = $self->ioloop;
   my $int  = $loop->recurring(1 => sub { });
   local $SIG{INT} = local $SIG{TERM} = sub { $loop->stop };
+  $self->app->worker_starting;
```

Comment:
This looks like the wrong place, not all `Mojo::Server::Daemon` instances are started with `run`.

Reply:
For example, this would not work for `Mojo::UserAgent::Server`, which all our tests rely on. That's a blocker for this PR i'm afraid.

Reply:
So what's the proper place for the hook to be fired?

Reply:
@kraih it does work with Mojo::UserAgent::Server now.